sdk/core/azure-core/src/main/java/com/azure/core/annotation/FormData.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotation for form parameters to be sent to a REST API Request URI.
+ *
+ * <p><strong>Example 1:</strong></p>
+ *
+ * <!-- src_embed com.azure.core.annotation.FormData.class -->
+ * <pre>
+ * &#64;Post&#40;&quot;spellcheck&quot;&#41;
+ * Mono&lt;Response&lt;ResponseBody&gt;&gt; spellChecker&#40&#64;HeaderParam&#40&quot;X-BingApis-SDK&quot;&#41; String xBingApisSDK,
+ *     &#64;QueryParam&#40&quot;UserId&quot;&#41; String userId,
+ *     &#64;FormData&#40&quot;Text&quot;&#41; String text&#41;;</pre>
+ * <!-- end com.azure.core.annotation.FormData.class -->
+ */
+@Retention(RUNTIME)
+@Target(PARAMETER)
+public @interface FormData {
+    /**
+     * @return The name of the key in a key value pair as part of the form data.
+     */
+    String value();
+
+    /**
+     * Optional value to be provided if {@link FormData#value()} represents a file's contents.
+     *
+     * @return The name of the file that {@link FormData#value()} represents.
+     */
+    String filename() default "";
```

Comment:
This needs to be changed as the annotation won't allow for customers to pass their own file names as this will be a compiled constant.

Reply:
The tests show passing a filename is possible, how would it not allow users to set such a value? Could you be clearer as to how this could be changed?

Reply:
The file name passing being shown in the testing is static, so in the service client if you had (as used in testing):

```java
void multipleFormDataBodyWithFilename(@FormData("name") String name,
    @FormData(value = "file", filename = "file.txt") String file,
    @FormData(value = "image", filename = "image.png") String image)
```

The `String file` will always be named `file.txt` and the `String image` will always be `image.png`. This is due to them being a property in the annotation which is an immutable constant. If the service needed to provide a way for the consumer to pass the file name it wouldn't be possible.

This is an area that still needs more investigation as when we do have a file in the `multipart/form-data` we need a way to pass both the contents and the name. This may take the form of a new type or by having multiple parameters exposed. Either way, it'll need to be something the annotation can handle.

Reply:
Thank you for the thorough explanation, I had misinterpreted what you said as the empty string value being static and not able to be changed to any other value.